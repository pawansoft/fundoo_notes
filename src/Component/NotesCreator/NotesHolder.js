import React, { Component } from 'react';
import AsyncStorage from '@react-native-async-storage/async-storage';
import{
    ScrollView,
    Text,
    TextInput,
    View
} from 'react-native';
import { Appbar } from 'react-native-paper';
import Textarea from 'react-native-textarea';
import FirebaseService from '../../../Services/firebase_services/FirebaseService';
import NotesHolderStyle from '../../Style/NotesHolderStyle';

export default class NotesHolder extends Component{
    constructor(props){
        super(props)
        this.state = {
            title : '',
            note : '', 
            nullValueError : '',
        }
    }

    handleTitle = async(title) => {
        await this.setState({
            title : title
        })
        console.log(this.state.title);
    }

    handleNote = async(note) => {
        await this.setState({
            note : note
        })
        console.log(this.state.note);
    }

    handleBackButton = async() => {
        const userId = await AsyncStorage.getItem('userId');
        if(this.state.title != null && this.state.note !=null){
            FirebaseService._storeNoteService(userId, this.state.title, this.state.note)
            .then(() => this.props.navigation.navigate('Notes'))
            .catch(error => console.log(error))
        }else{
            console.log('Fill the details');
        }
        
    }

    render(){
        return(
            <View style = {{flex :1}}>
               
                <Appbar style = {NotesHolderStyle.headerContainer}>
                    <View style = {{flexDirection : 'row', width : '100%', justifyContent : "space-around"}}>
                        <Appbar.Action
                        icon = 'keyboard-backspace'
                        onPress = {this.handleBackButton}
                        />

                        <Appbar.Action
                        icon = 'pin-outline'
                        />

                        <Appbar.Action
                        icon = 'bell-plus-outline'
                        />

                        <Appbar.Action
                        icon = 'archive-arrow-down-outline'
                        />
                    </View>
                </Appbar>
              
                <ScrollView>
                    <View style = {NotesHolderStyle.Title}>
                        <TextInput 
                        style = {NotesHolderStyle.TitleFont}
                        onChangeText = {this.handleTitle}
                        placeholder = 'Title'/>
                    </View>
                    <View style = {NotesHolderStyle.Note}>
                        <Textarea
                        placeholder = 'Notes'
                        onChangeText = {this.handleNote}/>
                    </View>
                </ScrollView>
                <Appbar style = {NotesHolderStyle.footerContainer}>
                <View style = {{flexDirection : 'row', width : '100%', justifyContent : "space-around"}}>
                        <Appbar.Action
                        icon = 'plus-box-outline'/>

                        <Appbar.Action
                        icon = 'undo-variant'/>

                        <Appbar.Action
                        icon = 'redo-variant'/>

                        <Appbar.Action
                        icon = 'dots-vertical'/>
                    </View>
                </Appbar>
            </View>
        )
    }
}