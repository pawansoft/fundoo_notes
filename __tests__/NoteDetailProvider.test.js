import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import NotesHolder from '../src/Component/NotesCreator/NotesHolder';
import { Appbar } from 'react-native-paper';

configure({adapter: new Adapter()})

jest.mock('react-native-localization', () => class RNLocalization {
    language = 'en-US'
  
    constructor (props) {
      this.props = props
      this.setLanguage(this.language)
    }
  
    setLanguage (interfaceLanguage) {
      this.language = interfaceLanguage
      if (this.props[interfaceLanguage]) {
        var localizedStrings = this.props[this.language]
        for (var key in localizedStrings) {
          if (localizedStrings.hasOwnProperty(key)) this[key] = localizedStrings[key]
        }
      }
    }
})
 describe('Test Note Detail Provider', () => {
     it('ProvideNoteDetaitProvider_WhenSnapMatches_TestShouldPass', () => {
         const component = shallow(<NotesHolder/>);
         expect(component).toMatchSnapshot();
     })
 })
 describe('Test Note Detail State', () => {
   it('ProvideDetail_WhenPassedToTextBox_ShouldStoreInState', async() => {
     const compnent = shallow(<NotesHolder/>);
    await compnent.instance().handleTitle('First Note');
     expect(compnent.instance().state.title).toBe('First Note');
   })

   it('ProvideDetail_WhenPassedToNote_ShouldStoreInNoteState', async() => {
    const compnent = shallow(<NotesHolder/>);
    await compnent.instance().handleNote('Hello World');
    expect(compnent.instance().state.note).toBe('Hello World');
   })
 })

 describe('Test Icons and Appbar', () => {
   it('ProvideAppbar_WhenCount_ShoudReturn8icon ', () => {
     const component = shallow(<NotesHolder/>);
     expect(component.find(Appbar.Action)).toHaveLength(8)
   })

   it('ProvideNoteHolder_WhenFindLogo_ShouldHaveOnScreen', () => {
     const component = shallow (<NotesHolder/>);
     expect(component.find(Appbar.Action).at(0).props().icon).toEqual('keyboard-backspace');
     expect(component.find(Appbar.Action).at(1).props().icon).toEqual('pin-outline');
     expect(component.find(Appbar.Action).at(2).props().icon).toEqual('bell-plus-outline');
     expect(component.find(Appbar.Action).at(3).props().icon).toEqual('archive-arrow-down-outline');
     expect(component.find(Appbar.Action).at(4).props().icon).toEqual('plus-box-outline');
     expect(component.find(Appbar.Action).at(5).props().icon).toEqual('undo-variant');
     expect(component.find(Appbar.Action).at(6).props().icon).toEqual('redo-variant');
     expect(component.find(Appbar.Action).at(7).props().icon).toEqual('dots-vertical');
     
   })
 })