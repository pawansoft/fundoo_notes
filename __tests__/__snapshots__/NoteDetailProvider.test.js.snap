// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Note Detail Provider ProvideNoteDetaitProvider_WhenSnapMatches_TestShouldPass 1`] = `ShallowWrapper {}`;
